packages/app/src/hooks/useSyncCurrentStateIntoOpenedProjects.ts

```diff
@@ -53,28 +53,48 @@ export function useSyncCurrentStateIntoOpenedProjects() {
       },
     }));
   }, [currentProject]);
+  
+  // Track project and graph state, so that when the user switches projects, we can track that state without saving the project.
+  const [prevProjectState, setPrevProjectState] = useState({
+    project: currentProject,
+    openedGraph: currentGraph?.metadata?.id,
+  });
+  useEffect(() => {
+    if (
+      currentGraph.metadata?.id != null && currentProject.graphs[currentGraph.metadata.id] && 
+      prevProjectState.project.metadata.id === currentProject.metadata.id
+    ) {
+      setPrevProjectState({
+        project: {
+          ...currentProject,
+          graphs: {
+            ...currentProject.graphs,
+            [currentGraph.metadata!.id!]: currentGraph,
+          }
+        },
+        openedGraph: currentGraph.metadata!.id!,
+      });
+    }
+  }, [currentGraph]);
```

Comment:
Shouldn't this include `currentProject` too? Technically it should also include `prevProjectState` if we're doing rules-of-hooks

Reply:
Almost certainly, yes. It didn't seem to matter when I was testing locally.

I've added the deps! No performance implication, and local testing looks good.